src/internal/target-runner.csx

```diff
@@ -60,6 +57,44 @@ public static class SimpleTargetsTargetRunner
         }
     }
 
+    private static void ValidateDependencies(IDictionary<string, Target> targets)
+    {
+        var missingDependencies = new SortedDictionary<string, ISet<string>>();
+
+        foreach (var targetEntry in targets)
+        {
+            foreach (var dependencyName in targetEntry.Value.Dependencies)
+            {
+                if (!targets.ContainsKey(dependencyName))
+                {
+                    var missingDependenciesForTarget = GetOrAddEmpty(missingDependencies, dependencyName);
+                    missingDependenciesForTarget.Add(targetEntry.Key);
+                }
+            }
+        }
+
+        if (!missingDependencies.Any())
+        {
+            return;
+        }
+
+        var manyDependencies = missingDependencies.Count() > 1;
+        var message = new StringBuilder($"Missing {(manyDependencies ? "dependencies" : "dependency")} detected:");
+        foreach (var missingDependencyEntry in missingDependencies)
+        {
+            var targetsCsv = string.Join(", ", missingDependencyEntry.Value.Select(target => $"\"{target.Replace("\"", "\"\"")}\""));
+            message.Append($@"\r\n  ""{missingDependencyEntry.Key.Replace("\"", "\"\"")}"", required by {targetsCsv}");

```

Comment:
This should not be a literal string, i.e. using the `@` sign. We want the `\r\n` to actually be CRLF, rather than the string `"\r\n"`. Or rather, we should probably be using `System.Environment.NewLine`.

Reply:
Ha. Good catch. Embedding `\r\n` inside a @-string. How embarrassing.